handler/handler_test.go

```diff
@@ -36,19 +37,116 @@ var (
 		UploadBucketName: testBucket,
 		VaultPath:        testVaultPath,
 	}
-	testCsvBody        = bufio.NewReader(bytes.NewReader([]byte("a,b,c,d,e,f,g,h,i,j,k,l")))
-	testCsvFileContent = bufio.NewReadWriter(testCsvBody, nil)
-	testPsk            = []byte{0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15}
-	errS3              = errors.New("test S3Upload error")
-	errVault           = errors.New("test Vault error")
-	errPsk             = errors.New("test PSK error")
-	errDataset         = errors.New("test DatasetAPI error")
+	testCsvBody = bufio.NewReader(bytes.NewReader([]byte("a,b,c,d,e,f,g,h,i,j,k,l")))
+	testPsk     = []byte{0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15}
+	errS3       = errors.New("test S3Upload error")
+	errVault    = errors.New("test Vault error")
+	errPsk      = errors.New("test PSK error")
+	errDataset  = errors.New("test DatasetAPI error")
 )
 
 var originalCreatePSK = handler.CreatePSK
 
 var ctx = context.Background()
 
+func TestValidateQueryResponse(t *testing.T) {
+	Convey("Given an empty handler", t, func() {
+		eventHandler := handler.NewInstanceComplete(testCfg, nil, nil, nil, nil, nil)
+
+		Convey("A valid Cantabular response is successfully validated", func() {
+			err := eventHandler.ValidateQueryResponse(cantabularResp())
+			So(err, ShouldBeNil)
+		})
+
+		Convey("Validating a nil cantabular response fails with the expected error", func() {
+			err := eventHandler.ValidateQueryResponse(nil)
+			So(err, ShouldResemble, errors.New("nil response"))
```

Comment:
It's often said that comparing error strings is generally a bad idea/anti-pattern and there are probably better ways to go about things.. Not the end of the world or suggesting any particular change here but something to think about in general

Reply:
I think I did this in different places (not only in this repo...) maybe we should revisit this in the future